spacy/pipeline/span_ruler.py

```diff
@@ -0,0 +1,345 @@
+            passed in automatically from the factory when the component is
+            added. Used to disable the current span ruler while creating
+            phrase patterns with the nlp object.
+        spans_key (str): The spans key to save the spans under. Defaults to
+            "ruler".
+        phrase_matcher_attr (int / str): Token attribute to match on, passed
+            to the internal PhraseMatcher as `attr`
+        validate (bool): Whether patterns should be validated, passed to
+            Matcher and PhraseMatcher as `validate`
+        patterns (iterable): Optional patterns to load in.
+        overwrite (bool): Whether to reset any existing spans under this spans
+            key.
+        scorer (Optional[Callable]): The scoring method. Defaults to
+            spacy.pipeline.spancat.spancat_score.
+
+        DOCS: https://spacy.io/api/spanruler#init
+        """
+        self.nlp = nlp
+        self.name = name
+        self.cfg = {
+            "spans_key": spans_key,
+            "overwrite": overwrite,
+            "phrase_matcher_attr": phrase_matcher_attr,
+            "validate": validate,
+        }
+        self.clear()
+        if patterns:
+            self.add_patterns(patterns)
+        self.scorer = scorer
+
+    def __len__(self) -> int:
+        """The number of all labels added to the span ruler."""
+        return len(set([p["label"] for p in self._patterns]))
+
+    def __contains__(self, label: str) -> bool:
+        """Whether a label is present in the patterns."""
+        return label in self.matcher or label in self.phrase_matcher
+
+    def __call__(self, doc: Doc) -> Doc:
+        """Find matches in document and add them as entities.
+
+        doc (Doc): The Doc object in the pipeline.
+        RETURNS (Doc): The Doc with added entities, if available.
+
+        DOCS: https://spacy.io/api/spanruler#call
+        """
+        error_handler = self.get_error_handler()
+        try:
+            matches = self.match(doc)
+            self.set_annotations(doc, matches)
+            return doc
+        except Exception as e:
+            return error_handler(self.name, self, [doc], e)
+
+    def match(self, doc: Doc):
+        self._require_patterns()
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="\\[W036")
+            matches = cast(
+                List[Tuple[int, int, int]],
+                list(self.matcher(doc)) + list(self.phrase_matcher(doc)),
+            )
+        deduplicated_matches = set(
+            (m_id, start, end) for m_id, start, end in matches if start != end
+        )
+        get_sort_key = lambda m: (m[2] - m[1], m[1])
+        final_matches = sorted(deduplicated_matches, key=get_sort_key)
+        return final_matches
+
+    def set_annotations(self, doc, matches):
+        """Modify the document in place"""
+        if not self.cfg["spans_key"] in doc.spans:
+            doc.spans[self.cfg["spans_key"]] = []
```

Comment:
We can have a property that accesses `self.cfg["spans_key"]`- for `spancat` it's just `key` so maybe we should take the same?

Reply:
Sure.